File: backend/api/models/disease.py
from django.db import models
from django.utils import timezone


class ExternalDetection(models.Model):
    """
    Stores official disease detections from CDFA, USDA, or other authorities.
    Used for proximity alerting to warn growers of nearby threats.
    """

    SOURCE_CHOICES = [
        ('cdfa', 'California Dept of Food & Agriculture'),
        ('usda', 'USDA APHIS'),
        ('county_ag', 'County Agricultural Commissioner'),
        ('uc_anr', 'UC Agriculture & Natural Resources'),
        ('manual', 'Manual Entry'),
    ]

    DISEASE_TYPE_CHOICES = [
        ('hlb', 'Huanglongbing (Citrus Greening)'),
        ('acp', 'Asian Citrus Psyllid'),
        ('ctvd', 'Citrus Tristeza Virus'),
        ('cyvcv', 'Citrus Yellow Vein Clearing Virus'),
        ('canker', 'Citrus Canker'),
        ('phytophthora', 'Phytophthora Root Rot'),
        ('laurel_wilt', 'Laurel Wilt'),
        ('other', 'Other'),
    ]

    LOCATION_TYPE_CHOICES = [
        ('residential', 'Residential/Backyard'),
        ('commercial', 'Commercial Grove'),
        ('nursery', 'Nursery'),
        ('unknown', 'Unknown'),
    ]

    # Source tracking
    source = models.CharField(max_length=50, choices=SOURCE_CHOICES)
    source_id = models.CharField(max_length=100, blank=True)

    # Disease info
    disease_type = models.CharField(max_length=50, choices=DISEASE_TYPE_CHOICES)
    disease_name = models.CharField(max_length=200)

    # Location
    latitude = models.DecimalField(max_digits=10, decimal_places=7)
    longitude = models.DecimalField(max_digits=10, decimal_places=7)
    county = models.CharField(max_length=100)
    city = models.CharField(max_length=100, blank=True)
    location_type = models.CharField(
        max_length=50,
        choices=LOCATION_TYPE_CHOICES,
        default='unknown'
    )

    # Dates
    detection_date = models.DateField()
    reported_date = models.DateField()
    fetched_at = models.DateTimeField(auto_now_add=True)

    # Status
    is_active = models.BooleanField(default=True)
    eradication_date = models.DateField(null=True, blank=True)

    # Metadata
    notes = models.TextField(blank=True)
    raw_data = models.JSONField(default=dict)

    class Meta:
        ordering = ['-detection_date']
        verbose_name = "External Detection"
        verbose_name_plural = "External Detections"
        indexes = [
            models.Index(fields=['disease_type', 'is_active'], name='idx_extdet_disease_active'),
            models.Index(fields=['latitude', 'longitude'], name='idx_extdet_coords'),
            models.Index(fields=['county'], name='idx_extdet_county'),
        ]
        constraints = [
            models.UniqueConstraint(
                fields=['source', 'source_id'],
                name='unique_external_detection_source',
                condition=models.Q(source_id__gt=''),
            ),
        ]

    def __str__(self):
        return f"{self.disease_name} - {self.county} ({self.detection_date})"


class QuarantineZone(models.Model):
    """
    CDFA quarantine boundary polygons.

    Stores geographic boundaries of quarantine zones for HLB, ACP, and other
    diseases/pests. Used for visualization on threat maps and compliance checking.
    """

    ZONE_TYPE_CHOICES = [
        ('hlb', 'HLB Quarantine'),
        ('acp', 'ACP Quarantine'),
        ('eradication', 'Eradication Area'),
        ('buffer', 'Buffer Zone'),
        ('other', 'Other'),
    ]

    zone_type = models.CharField(max_length=50, choices=ZONE_TYPE_CHOICES)
    name = models.CharField(max_length=200)
    description = models.TextField(blank=True)

    # GeoJSON Polygon boundary
    boundary = models.JSONField(
        help_text="GeoJSON Polygon geometry for the quarantine boundary"
    )

    # Source tracking
    source = models.CharField(max_length=100, default='cdfa')
    source_url = models.URLField(blank=True)
    source_id = models.CharField(max_length=100, blank=True)

    # Dates
    established_date = models.DateField()
    last_updated = models.DateField(auto_now=True)
    expires_date = models.DateField(null=True, blank=True)

    # Status
    is_active = models.BooleanField(default=True)

    # Geographic info
    county = models.CharField(max_length=100, blank=True)
    state = models.CharField(max_length=50, default='California')

    # Timestamps
    created_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)

    class Meta:
        ordering = ['-established_date']
        verbose_name = "Quarantine Zone"
        verbose_name_plural = "Quarantine Zones"
        indexes = [
            models.Index(fields=['zone_type', 'is_active'], name='idx_qzone_type_active'),
            models.Index(fields=['county'], name='idx_qzone_county'),
        ]
        constraints = [
            models.UniqueConstraint(
                fields=['source', 'source_id'],
                name='unique_quarantine_zone_source',
                condition=models.Q(source_id__gt=''),
            ),
        ]

    def __str__(self):
        return f"{self.name} ({self.zone_type.upper()})"


class DiseaseAlertRule(models.Model):
    """
    Configurable rules for generating disease alerts.
    Allows per-company customization of alert thresholds.
    """

    RULE_TYPE_CHOICES = [
        ('proximity', 'Proximity Alert'),
        ('ndvi_threshold', 'NDVI Threshold'),
        ('ndvi_change', 'NDVI Change Rate'),
        ('canopy_loss', 'Canopy Loss'),
        ('tree_count_change', 'Tree Count Change'),
        ('regional_trend', 'Regional Trend'),
    ]

    PRIORITY_CHOICES = [
        ('critical', 'Critical'),
        ('high', 'High'),
        ('medium', 'Medium'),
        ('low', 'Low'),
    ]

    company = models.ForeignKey(
        'Company',
        on_delete=models.CASCADE,
        related_name='disease_alert_rules'
    )

    name = models.CharField(max_length=100)
    description = models.TextField(blank=True)
    rule_type = models.CharField(max_length=50, choices=RULE_TYPE_CHOICES)
    conditions = models.JSONField(default=dict)
    alert_priority = models.CharField(max_length=20, choices=PRIORITY_CHOICES)

    send_email = models.BooleanField(default=True)
    send_sms = models.BooleanField(default=False)
    send_immediately = models.BooleanField(default=False)

    is_active = models.BooleanField(default=True)
    created_at = models.DateTimeField(auto_now_add=True)
    created_by = models.ForeignKey(
        'User',
        on_delete=models.SET_NULL,
        null=True,
        blank=True,
        related_name='created_disease_rules'
    )

    class Meta:
        ordering = ['name']
        verbose_name = "Disease Alert Rule"
        verbose_name_plural = "Disease Alert Rules"

    def __str__(self):
        return f"{self.name} ({self.get_rule_type_display()})"


class DiseaseAnalysisRun(models.Model):
    """
    Tracks a disease/health analysis run for a field.
    Parallel to TreeDetectionRun but focused on health trends.
    """

    STATUS_CHOICES = [
        ('pending', 'Pending'),
        ('processing', 'Processing'),
        ('completed', 'Completed'),
        ('failed', 'Failed'),
    ]

    ANALYSIS_TYPE_CHOICES = [
        ('ndvi_trend', 'NDVI Trend Analysis'),
        ('anomaly_detection', 'Anomaly Detection'),
        ('full', 'Full Health Analysis'),
    ]

    RISK_LEVEL_CHOICES = [
        ('low', 'Low'),
        ('moderate', 'Moderate'),
        ('high', 'High'),
        ('critical', 'Critical'),
    ]

    company = models.ForeignKey(
        'Company',
        on_delete=models.CASCADE,
        related_name='disease_analyses'
    )
    field = models.ForeignKey(
        'Field',
        on_delete=models.CASCADE,
        related_name='disease_analyses'
    )

    # Processing status
    status = models.CharField(max_length=20, choices=STATUS_CHOICES, default='pending')
    error_message = models.TextField(blank=True)

    # Analysis parameters
    analysis_type = models.CharField(
        max_length=50,
        choices=ANALYSIS_TYPE_CHOICES,
        default='full'
    )
    parameters = models.JSONField(default=dict)

    # Results - Field Level
    avg_ndvi = models.DecimalField(max_digits=4, decimal_places=3, null=True)
    ndvi_change_30d = models.DecimalField(max_digits=5, decimal_places=3, null=True)
    ndvi_change_90d = models.DecimalField(max_digits=5, decimal_places=3, null=True)
    canopy_coverage_percent = models.DecimalField(max_digits=5, decimal_places=2, null=True)
    canopy_change_30d = models.DecimalField(max_digits=5, decimal_places=2, null=True)

    # Results - Tree Level Aggregates
    total_trees_analyzed = models.IntegerField(default=0)
    trees_healthy = models.IntegerField(default=0)
    trees_mild_stress = models.IntegerField(default=0)
    trees_moderate_stress = models.IntegerField(default=0)
    trees_severe_stress = models.IntegerField(default=0)
    trees_declining = models.IntegerField(default=0)

    # Risk Assessment
    health_score = models.IntegerField(null=True)
    risk_level = models.CharField(
        max_length=20,
        choices=RISK_LEVEL_CHOICES,
        null=True,
        blank=True
    )
    risk_factors = models.JSONField(default=list)

    # Anomaly Detection
    anomaly_zones = models.JSONField(default=list)
    anomaly_count = models.IntegerField(default=0)

    # Recommendations
    recommendations = models.JSONField(default=list)

    # Timestamps
    created_at = models.DateTimeField(auto_now_add=True)
    completed_at = models.DateTimeField(null=True)

    # Review
    reviewed_by = models.ForeignKey(
        'User',
        on_delete=models.SET_NULL,
        null=True,
        blank=True,
        related_name='reviewed_disease_analyses'
    )
    review_notes = models.TextField(blank=True)

    class Meta:
        ordering = ['-created_at']
        verbose_name = "Disease Analysis Run"
        verbose_name_plural = "Disease Analysis Runs"
        indexes = [
            models.Index(fields=['company', 'status'], name='idx_disanalysis_co_status'),
            models.Index(fields=['field', '-created_at'], name='idx_disanalysis_field_date'),
        ]

    def __str__(self):
        return f"{self.field.name} - {self.created_at.strftime('%Y-%m-%d')} ({self.status})"


class DiseaseAlert(models.Model):
    """
    Disease-specific alerts for users.
    Follows ComplianceAlert pattern but with disease-specific fields.
    """

    ALERT_TYPE_CHOICES = [
        ('proximity_hlb', 'HLB Detected Nearby'),
        ('proximity_acp', 'ACP Activity Nearby'),
        ('proximity_other', 'Other Disease Nearby'),
        ('ndvi_anomaly', 'NDVI Anomaly Detected'),
        ('tree_decline', 'Tree Decline Detected'),
        ('canopy_loss', 'Canopy Loss Detected'),
        ('regional_trend', 'Regional Health Trend'),
        ('scouting_verified', 'Verified Scouting Report'),
    ]

    PRIORITY_CHOICES = [
        ('critical', 'Critical'),
        ('high', 'High'),
        ('medium', 'Medium'),
        ('low', 'Low'),
    ]

    company = models.ForeignKey(
        'Company',
        on_delete=models.CASCADE,
        related_name='disease_alerts'
    )
    farm = models.ForeignKey(
        'Farm',
        on_delete=models.CASCADE,
        null=True,
        blank=True,
        related_name='disease_alerts'
    )
    field = models.ForeignKey(
        'Field',
        on_delete=models.CASCADE,
        null=True,
        blank=True,
        related_name='disease_alerts'
    )

    alert_type = models.CharField(max_length=50, choices=ALERT_TYPE_CHOICES)
    priority = models.CharField(max_length=20, choices=PRIORITY_CHOICES)
    title = models.CharField(max_length=200)
    message = models.TextField()

    # Context
    distance_miles = models.DecimalField(max_digits=6, decimal_places=2, null=True, blank=True)
    related_detection = models.ForeignKey(
        ExternalDetection,
        on_delete=models.SET_NULL,
        null=True,
        blank=True,
        related_name='alerts'
    )
    related_analysis = models.ForeignKey(
        DiseaseAnalysisRun,
        on_delete=models.SET_NULL,
        null=True,
        blank=True,
        related_name='alerts'
    )

    # Actions
    recommended_actions = models.JSONField(default=list)
    action_url = models.CharField(max_length=500, blank=True)
    action_label = models.CharField(max_length=100, blank=True)

    # Status
    is_active = models.BooleanField(default=True)
    is_acknowledged = models.BooleanField(default=False)
    acknowledged_by = models.ForeignKey(
        'User',
        on_delete=models.SET_NULL,
        null=True,
        blank=True,
        related_name='acknowledged_disease_alerts'
    )
    acknowledged_at = models.DateTimeField(null=True, blank=True)

    # Notifications sent
    email_sent = models.BooleanField(default=False)
    email_sent_at = models.DateTimeField(null=True, blank=True)
    sms_sent = models.BooleanField(default=False)
    sms_sent_at = models.DateTimeField(null=True, blank=True)

    # Timestamps
    created_at = models.DateTimeField(auto_now_add=True)
    expires_at = models.DateTimeField(null=True, blank=True)

    class Meta:
        ordering = ['-created_at']
        verbose_name = "Disease Alert"
        verbose_name_plural = "Disease Alerts"
        indexes = [
            models.Index(fields=['company', 'is_active', 'priority'], name='idx_disalert_co_active_pri'),
            models.Index(fields=['alert_type', 'created_at'], name='idx_disalert_type_created'),
        ]

    def __str__(self):
        return f"[{self.get_priority_display()}] {self.title}"

    def acknowledge(self, user):
        """Acknowledge this alert."""
        self.is_acknowledged = True
        self.acknowledged_by = user
        self.acknowledged_at = timezone.now()
        self.save()

    def dismiss(self):
        """Dismiss/deactivate this alert."""
        self.is_active = False
        self.save()


class ScoutingReport(models.Model):
    """
    User-submitted disease/pest observations for crowdsourced monitoring.
    """

    REPORT_TYPE_CHOICES = [
        ('disease_symptom', 'Disease Symptom'),
        ('pest_sighting', 'Pest Sighting'),
        ('tree_decline', 'Tree Decline'),
        ('tree_death', 'Tree Death'),
        ('acp_sighting', 'Asian Citrus Psyllid'),
        ('other', 'Other'),
    ]

    SEVERITY_CHOICES = [
        ('low', 'Low - Minor/Isolated'),
        ('medium', 'Medium - Several Trees'),
        ('high', 'High - Significant Area'),
    ]

    AI_STATUS_CHOICES = [
        ('pending', 'Pending'),
        ('processing', 'Processing'),
        ('completed', 'Completed'),
        ('skipped', 'Skipped'),
    ]

    STATUS_CHOICES = [
        ('submitted', 'Submitted'),
        ('under_review', 'Under Review'),
        ('verified', 'Verified'),
        ('false_alarm', 'False Alarm'),
        ('inconclusive', 'Inconclusive'),
    ]

    company = models.ForeignKey(
        'Company',
        on_delete=models.CASCADE,
        related_name='scouting_reports'
    )
    reported_by = models.ForeignKey(
        'User',
        on_delete=models.CASCADE,
        related_name='scouting_reports'
    )
    farm = models.ForeignKey(
        'Farm',
        on_delete=models.CASCADE,
        null=True,
        blank=True,
        related_name='scouting_reports'
    )
    field = models.ForeignKey(
        'Field',
        on_delete=models.CASCADE,
        null=True,
        blank=True,
        related_name='scouting_reports'
    )

    # Location
    latitude = models.DecimalField(max_digits=10, decimal_places=7)
    longitude = models.DecimalField(max_digits=10, decimal_places=7)

    # Report details
    report_type = models.CharField(max_length=50, choices=REPORT_TYPE_CHOICES)
    symptoms = models.JSONField(default=dict)
    severity = models.CharField(max_length=20, choices=SEVERITY_CHOICES)
    affected_tree_count = models.IntegerField(null=True, blank=True)
    notes = models.TextField(blank=True)

    # AI Analysis
    ai_analysis_status = models.CharField(
        max_length=20,
        choices=AI_STATUS_CHOICES,
        default='pending'
    )
    ai_diagnosis = models.JSONField(default=dict)

    # Verification
    status = models.CharField(max_length=20, choices=STATUS_CHOICES, default='submitted')
    verified_by = models.ForeignKey(
        'User',
        on_delete=models.SET_NULL,
        null=True,
        blank=True,
        related_name='verified_scouting_reports'
    )
    verification_notes = models.TextField(blank=True)

    # Sharing
    share_anonymously = models.BooleanField(default=True)
    is_public = models.BooleanField(default=False)

    # Timestamps
    observed_date = models.DateField()
    created_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)

    class Meta:
        ordering = ['-created_at']
        verbose_name = "Scouting Report"
        verbose_name_plural = "Scouting Reports"
        indexes = [
            models.Index(fields=['farm', '-observed_date'], name='idx_scout_farm_date'),
            models.Index(fields=['field', '-observed_date'], name='idx_scout_field_date'),
        ]

    def __str__(self):
        return f"{self.get_report_type_display()} - {self.observed_date}"


class ScoutingPhoto(models.Model):
    """Photos attached to scouting reports."""

    report = models.ForeignKey(
        ScoutingReport,
        on_delete=models.CASCADE,
        related_name='photos'
    )
    image = models.ImageField(upload_to='scouting/%Y/%m/')
    caption = models.CharField(max_length=200, blank=True)
    uploaded_at = models.DateTimeField(auto_now_add=True)

    class Meta:
        verbose_name = "Scouting Photo"
        verbose_name_plural = "Scouting Photos"

    def __str__(self):
        return f"Photo for {self.report}"


class TreeHealthRecord(models.Model):
    """
    Tracks health metrics for individual trees over time.
    Links DetectedTree records across multiple detection runs.
    """

    TREND_CHOICES = [
        ('improving', 'Improving'),
        ('stable', 'Stable'),
        ('declining', 'Declining'),
        ('rapid_decline', 'Rapid Decline'),
    ]

    HEALTH_STATUS_CHOICES = [
        ('healthy', 'Healthy'),
        ('mild_stress', 'Mild Stress'),
        ('moderate_stress', 'Moderate Stress'),
        ('severe_stress', 'Severe Stress'),
        ('dead', 'Dead/Removed'),
    ]

    company = models.ForeignKey(
        'Company',
        on_delete=models.CASCADE,
        related_name='tree_health_records'
    )
    field = models.ForeignKey(
        'Field',
        on_delete=models.CASCADE,
        related_name='tree_health_records'
    )

    # Tree identification
    tree_id = models.CharField(max_length=50)
    latitude = models.DecimalField(max_digits=10, decimal_places=7)
    longitude = models.DecimalField(max_digits=10, decimal_places=7)

    # Current state
    current_ndvi = models.DecimalField(max_digits=4, decimal_places=3, null=True)
    current_canopy_diameter_m = models.DecimalField(max_digits=4, decimal_places=2, null=True)
    last_updated = models.DateTimeField(auto_now=True)

    # Trend data
    ndvi_history = models.JSONField(default=list)
    canopy_history = models.JSONField(default=list)
    ndvi_trend = models.CharField(max_length=20, choices=TREND_CHOICES, default='stable')

    # Health status
    health_status = models.CharField(
        max_length=20,
        choices=HEALTH_STATUS_CHOICES,
        default='healthy'
    )

    # Flags
    flagged_for_inspection = models.BooleanField(default=False)
    flag_reason = models.TextField(blank=True)
    inspected = models.BooleanField(default=False)
    inspection_date = models.DateField(null=True, blank=True)
    inspection_notes = models.TextField(blank=True)

    class Meta:
        verbose_name = "Tree Health Record"
        verbose_name_plural = "Tree Health Records"
        constraints = [
            models.UniqueConstraint(
                fields=['field', 'tree_id'],
                name='unique_tree_health_field_treeid'
            ),
        ]
        indexes = [
            models.Index(fields=['field', 'health_status'], name='idx_treehealth_field_status'),
            models.Index(fields=['flagged_for_inspection'], name='idx_treehealth_flagged'),
        ]

    def __str__(self):
        return f"Tree {self.tree_id} - {self.field.name}"

File: frontend/src/services/api/disease.js
// =============================================================================
// DISEASE PREVENTION APIs
// =============================================================================

import api, { createCRUDAPI } from './index';

/**
 * External Detections - Official disease detections from CDFA, USDA, etc.
 */
export const externalDetectionsAPI = {
  ...createCRUDAPI('disease/external-detections'),
  sync: () => api.post('/disease/external-detections/sync/'),
  nearPoint: (lat, lng, radiusMiles = 15) =>
    api.get('/disease/external-detections/near_point/', {
      params: { latitude: lat, longitude: lng, radius_miles: radiusMiles },
    }),
};

/**
 * Disease Alerts - User-facing disease notifications
 */
export const diseaseAlertsAPI = {
  /** Get all alerts with optional filters */
  getAll: (params = {}) => api.get('/disease/alerts/', { params }),

  /** Get active alerts only */
  active: () => api.get('/disease/alerts/active/'),

  /** Get a specific alert */
  get: (id) => api.get(`/disease/alerts/${id}/`),

  /** Acknowledge an alert */
  acknowledge: (id) => api.post(`/disease/alerts/${id}/acknowledge/`),

  /** Dismiss an alert */
  dismiss: (id) => api.post(`/disease/alerts/${id}/dismiss/`),

  /** Get alert summary counts */
  summary: () => api.get('/disease/alerts/summary/'),

  /** Get alerts for a specific farm */
  byFarm: (farmId) => api.get('/disease/alerts/', { params: { farm: farmId } }),
};

/**
 * Disease Alert Rules - Configurable alert triggers
 */
export const diseaseAlertRulesAPI = {
  ...createCRUDAPI('disease/alert-rules'),
  toggle: (id, isActive) => api.patch(`/disease/alert-rules/${id}/`, { is_active: isActive }),
};

/**
 * Disease Analysis Runs - Field health analysis
 */
export const diseaseAnalysesAPI = {
  /** Get all analysis runs */
  getAll: (params = {}) => api.get('/disease/analyses/', { params }),

  /** Get a specific analysis run */
  get: (id) => api.get(`/disease/analyses/${id}/`),

  /** Get trees with health data for an analysis */
  getTrees: (id, params = {}) => api.get(`/disease/analyses/${id}/trees/`, { params }),

  /** Trigger analysis for a field */
  analyzeField: (fieldId, params = {}) =>
    api.post(`/disease/analyses/analyze_field/`, { field_id: fieldId, ...params }),
};

/**
 * Scouting Reports - Crowdsourced disease observations
 */
export const scoutingReportsAPI = {
  ...createCRUDAPI('disease/scouting'),
  verify: (id, data) => api.post(`/disease/scouting/${id}/verify/`, data),
  addPhoto: (id, formData) =>
    api.post(`/disease/scouting/${id}/photos/`, formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    }),
  byField: (fieldId) => api.get('/disease/scouting/', { params: { field: fieldId } }),
};

/**
 * Disease Dashboard - Unified disease prevention overview
 */
export const diseaseDashboardAPI = {
  /** Get dashboard data */
  get: () => api.get('/disease/dashboard/'),

  /** Get proximity risks for company */
  getProximityRisks: () => api.get('/disease/dashboard/proximity_risks/'),

  /** Get company risk score */
  getRiskScore: () => api.get('/disease/dashboard/risk_score/'),

  /** Get regional threat map data */
  getRegionalData: (params = {}) => api.get('/disease/dashboard/regional/', { params }),

  /** Get threat map data (farms, detections, quarantine zones) */
  getMapData: () => api.get('/disease/dashboard/map_data/'),

  /** HLB risk per field (all fields by default, or ?field_id=X for one) */
  getHLBRisk: (params = {}) => api.get('/disease/dashboard/hlb-risk/', { params }),
};

/**
 * Combined Disease API - alias for backward compatibility and convenience
 */
export const diseaseAPI = {
  ...diseaseDashboardAPI,
  getMapData: () => api.get('/disease/dashboard/map_data/'),
};

/**
 * Field Health API - Field-specific health endpoints
 */
export const fieldHealthAPI = {
  /** Get health summary for a field */
  getSummary: (fieldId) => api.get(`/fields/${fieldId}/health/`),

  /** Get health history for a field */
  getHistory: (fieldId, params = {}) =>
    api.get(`/fields/${fieldId}/health/history/`, { params }),

  /** Trigger health analysis for a field */
  analyze: (fieldId) => api.post(`/fields/${fieldId}/health/analyze/`),

  /** Get trees with health data for a field */
  getTrees: (fieldId, params = {}) =>
    api.get(`/fields/${fieldId}/health/trees/`, { params }),
};

// Constants for Disease Prevention
export const DISEASE_CONSTANTS = {
  DISEASE_TYPES: [
    { value: 'hlb', label: 'Huanglongbing (Citrus Greening)', color: '#dc2626' },
    { value: 'acp', label: 'Asian Citrus Psyllid', color: '#f97316' },
    { value: 'ctvd', label: 'Citrus Tristeza Virus', color: '#eab308' },
    { value: 'cyvcv', label: 'Citrus Yellow Vein Clearing Virus', color: '#84cc16' },
    { value: 'canker', label: 'Citrus Canker', color: '#ef4444' },
    { value: 'phytophthora', label: 'Phytophthora Root Rot', color: '#8b5cf6' },
    { value: 'laurel_wilt', label: 'Laurel Wilt', color: '#6366f1' },
    { value: 'other', label: 'Other', color: '#6b7280' },
  ],

  HEALTH_STATUSES: [
    { value: 'healthy', label: 'Healthy', color: '#22c55e' },
    { value: 'mild_stress', label: 'Mild Stress', color: '#84cc16' },
    { value: 'moderate_stress', label: 'Moderate Stress', color: '#eab308' },
    { value: 'severe_stress', label: 'Severe Stress', color: '#f97316' },
  ],

  RISK_LEVELS: [
    { value: 'low', label: 'Low', color: '#22c55e' },
    { value: 'moderate', label: 'Moderate', color: '#eab308' },
    { value: 'high', label: 'High', color: '#f97316' },
    { value: 'critical', label: 'Critical', color: '#dc2626' },
  ],

  ALERT_TYPES: [
    { value: 'proximity_hlb', label: 'HLB Detected Nearby', icon: '🔴' },
    { value: 'proximity_acp', label: 'ACP Activity Nearby', icon: '🟠' },
    { value: 'proximity_other', label: 'Other Disease Nearby', icon: '🟡' },
    { value: 'ndvi_anomaly', label: 'NDVI Anomaly Detected', icon: '📉' },
    { value: 'tree_decline', label: 'Tree Decline Detected', icon: '🌳' },
    { value: 'canopy_loss', label: 'Canopy Loss Detected', icon: '🍂' },
    { value: 'regional_trend', label: 'Regional Health Trend', icon: '📊' },
    { value: 'scouting_verified', label: 'Verified Scouting Report', icon: '✅' },
  ],

  ALERT_PRIORITIES: [
    { value: 'critical', label: 'Critical', color: '#dc2626' },
    { value: 'high', label: 'High', color: '#f97316' },
    { value: 'medium', label: 'Medium', color: '#eab308' },
    { value: 'low', label: 'Low', color: '#3b82f6' },
  ],

  SCOUTING_REPORT_TYPES: [
    { value: 'disease_symptom', label: 'Disease Symptom' },
    { value: 'pest_sighting', label: 'Pest Sighting' },
    { value: 'tree_decline', label: 'Tree Decline' },
    { value: 'tree_death', label: 'Tree Death' },
    { value: 'acp_sighting', label: 'Asian Citrus Psyllid' },
    { value: 'other', label: 'Other' },
  ],

  SCOUTING_SEVERITIES: [
    { value: 'low', label: 'Low - Minor/Isolated', color: '#22c55e' },
    { value: 'medium', label: 'Medium - Several Trees', color: '#eab308' },
    { value: 'high', label: 'High - Significant Area', color: '#ef4444' },
  ],

  SCOUTING_STATUSES: [
    { value: 'submitted', label: 'Submitted', color: '#6b7280' },
    { value: 'under_review', label: 'Under Review', color: '#3b82f6' },
    { value: 'verified', label: 'Verified', color: '#22c55e' },
    { value: 'false_alarm', label: 'False Alarm', color: '#ef4444' },
    { value: 'inconclusive', label: 'Inconclusive', color: '#f59e0b' },
  ],

  // HLB Symptom Checklist
  SYMPTOM_CHECKLIST: [
    { key: 'yellowing_asymmetric', label: 'Yellowing of leaves (asymmetric/blotchy)' },
    { key: 'green_islands', label: 'Green islands on yellow leaves' },
    { key: 'fruit_lopsided', label: 'Lopsided or misshapen fruit' },
    { key: 'fruit_drop', label: 'Premature fruit drop' },
    { key: 'twig_dieback', label: 'Twig dieback' },
    { key: 'overall_decline', label: 'Tree in overall decline' },
    { key: 'acp_adult', label: 'Spotted Asian citrus psyllid (adult)' },
    { key: 'acp_nymph', label: 'Waxy psyllid nymphs on new growth' },
  ],

  DETECTION_SOURCES: [
    { value: 'cdfa', label: 'California Dept of Food & Agriculture' },
    { value: 'usda', label: 'USDA APHIS' },
    { value: 'county_ag', label: 'County Agricultural Commissioner' },
    { value: 'uc_anr', label: 'UC Agriculture & Natural Resources' },
    { value: 'manual', label: 'Manual Entry' },
  ],

  LOCATION_TYPES: [
    { value: 'residential', label: 'Residential/Backyard' },
    { value: 'commercial', label: 'Commercial Grove' },
    { value: 'nursery', label: 'Nursery' },
    { value: 'unknown', label: 'Unknown' },
  ],

  RULE_TYPES: [
    { value: 'proximity', label: 'Proximity Alert' },
    { value: 'ndvi_threshold', label: 'NDVI Threshold' },
    { value: 'ndvi_change', label: 'NDVI Change Rate' },
    { value: 'canopy_loss', label: 'Canopy Loss' },
    { value: 'tree_count_change', label: 'Tree Count Change' },
    { value: 'regional_trend', label: 'Regional Trend' },
  ],
};
